### PR TITLE
inline chat: fix leaks found in testing

### PR DESCRIPTION
**Pull request by connor4312**
Fixes some leaks I found while looking at https://github.com/microsoft/vscode/pull/190444, by adding ensureNoDisposablesAreLeakedInTestSuite to the suite.

I don't have a dev setup for inline chat, so I have not tested this beyond running tests and verifying the fix

### CODE REVIEW

File: src/vs/editor/browser/widget/diffEditorWidget2/diffEditorEditors.ts [hunk 1]
```diff
@@ -35,8 +35,8 @@ export class DiffEditorEditors extends Disposable {
 	) {
 		super();
 
-		this.original = this._createLeftHandSideEditor(_options.editorOptions.get(), codeEditorWidgetOptions.originalEditor || {});
-		this.modified = this._createRightHandSideEditor(_options.editorOptions.get(), codeEditorWidgetOptions.modifiedEditor || {});
+		this.original = this._register(this._createLeftHandSideEditor(_options.editorOptions.get(), codeEditorWidgetOptions.originalEditor || {}));
+		this.modified = this._register(this._createRightHandSideEditor(_options.editorOptions.get(), codeEditorWidgetOptions.modifiedEditor || {}));
```
Comment (connor4312) [inline src/vs/editor/browser/widget/diffEditorWidget2/diffEditorEditors.ts:39]: fyi @hediet

**Review by DonJayamanne — DISMISSED** [review]

**Review by hediet — DISMISSED** [review]

**Review by jrieken — DISMISSED** [review]
Thanks @connor4312

**Comment by jrieken** [comment]
@connor4312 tests are still failing

**Comment by connor4312** [comment]
Unfortunately these are the earlier test failures that also caused the merge editor tests to fail earlier. What I could do is merge this as an improvement with the leak detector commented out until we find the root cause

**Comment by connor4312** [comment]
I'll merge this in debt week

File: src/vs/workbench/contrib/inlineChat/test/browser/inlineChatController.test.ts [hunk 2]
```diff
@@ -145,7 +145,11 @@ suite('InteractiveChatController', function () {
 		ctrl?.dispose();
 	});
 
+	// todo: re-enable this when earlier tests are fixed
+	// ensureNoDisposablesAreLeakedInTestSuite();
+
 	test('creation, not showing anything', function () {
+		for (let deadline = Date.now() + 1000; Date.now() < deadline;) { }
```
Comment (rzhao271) [inline src/vs/workbench/contrib/inlineChat/test/browser/inlineChatController.test.ts:152]: Intentional?

**Review by jrieken — APPROVED** [review]

**Pull request merged**